core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -102,7 +102,8 @@ class TxnMarkerQueue(@volatile var destination: Node) {
   }
 
   def addMarkers(txnTopicPartition: Int, txnIdAndMarker: TxnIdAndMarkerEntry): Unit = {
-    val queue = markersPerTxnTopicPartition.getOrElseUpdate(txnTopicPartition, new LinkedBlockingQueue[TxnIdAndMarkerEntry]())
+    lazy val newQueue = new LinkedBlockingQueue[TxnIdAndMarkerEntry]()
```

Comment:
I'm not too familiar with lazy semantics, but would we not initialize this in `putIfAbsent` since that is the first access?

Reply:
Yes, no benefit in making it lazy

Reply:
Would it better to do `markersPerTxnTopicPartition.putIfAbsent(txnTopicPartition, new LinkedBlockingQueue[TxnIdAndMarkerEntry()).getOrElse(markersPerTxnTopicPartition.get(txnTopicPartition))`?

Reply:
To avoid the allocation, another way is to do a `get` first. If no null, then you can do the `putIfAbsent`.

Reply:
ok, will do that

Reply:
bq. to do a get first. If null,

There is a chance that the putIfAbsent() call would return Queue created by someone else (due to race).
It seems the extra check is not worthwhile.

Reply:
This is the code with the check:
```
map.get(key) match {
      case Some(value) => value
      case None =>
        val value = createValue
        map.putIfAbsent(key, value).getOrElse(value)
    }
```
Alternatively, I could do:
```
map.putIfAbsent(key, createValue).getOrElse(map.get(key))
```
I don't mind either way. @jason What do you think?

Reply:
I updated the code without the extra check btw.

Reply:
@tedyu I don't really get your comment. It's fine if the queue is created by someone else. We just need to ensure we get the instance that was inserted.

Reply:
Looking at latest change, there is no extra check.
Good by me.

Reply:
I had to update with the `get` first to ensure the object doesn't get created. But it does avoid the race too since it uses `putIfAbsent`.